src/transport/raw/PeerAddress.h

```diff
@@ -116,20 +116,30 @@ class PeerAddress
     bool operator==(const PeerAddress & other) const
     {
         // Compare common fields
-        if (mTransportType != other.mTransportType || mIPAddress != other.mIPAddress || mPort != other.mPort ||
-            mInterface != other.mInterface)
+        if (mTransportType != other.mTransportType)
         {
             return false;
         }
 
-        // Compare transport-type-specific fields
-        if (mTransportType == Type::kNfc)
+        // Compare transport-type specific fields
+        switch (mTransportType)
         {
-            return mId.mNFCShortId == other.mId.mNFCShortId;
-        }
+        case Type::kNfc:
+            return (mId.mNFCShortId == other.mId.mNFCShortId);
+
+        case Type::kUdp:
+        case Type::kTcp:
+            return (mIPAddress == other.mIPAddress && mPort == other.mPort && mInterface == other.mInterface);
 
-        // For other transport types, no additional fields to compare
-        return true;
+        case Type::kWiFiPAF:
+            return (mIPAddress == other.mIPAddress && mPort == other.mPort && mInterface == other.mInterface &&
+                    mId.mRemoteId == other.mId.mRemoteId);
```

Comment:
```suggestion
            return (mId.mRemoteId == other.mId.mRemoteId);
```
all those other fields are unused for PAF PeerAddress, no?

Reply:
Those are not used for PAF, since the device has not joint the network at that time.